reactor-core/src/main/java/reactor/util/retry/Retry.java

```diff
@@ -41,19 +54,19 @@
 	 * the attempt is delayed as well. This method generates the companion, out of a {@link Flux} of {@link RetrySignal},
 	 * which itself can serve as the simplest form of retry companion (indefinitely and immediately retry on any error).
 	 *
-	 * @param retrySignalCompanion the original {@link Flux} of {@link RetrySignal}, notifying of each source error that
-	 * _might_ result in a retry attempt, with context around the error and current retry cycle.
+	 * @param retrySignals the original {@link Flux} of {@link RetrySignal}, notifying of each source error that
+	 * <i>might</i> result in a retry attempt, with context around the error and current retry cycle.
 	 * @return the actual companion to use, which might delay or limit retry attempts
 	 */
-	Publisher<?> generateCompanion(Flux<RetrySignal> retrySignalCompanion);
+	public abstract Publisher<?> generateCompanion(Flux<RetrySignal> retrySignals);
```

Comment:
in an abstract class, you may want to have a `final` method that wraps the user-provided one, so that we keep _some_ control. Usually it makes it easier to make changes in future